app/helpers/govuk_form_helper.rb

```diff
@@ -0,0 +1,131 @@
+module GovukFormHelper
+  # = GOVUK Form Tag Helpers
+  #
+  # For usage, components and patterns see:
+  # https://design-system.service.gov.uk/
+
+  # Can be used to wrap multiple or single components.
+  #
+  # Creates a div with the 'govuk-form-group' css class, if the provided object has errors, adds
+  # the 'govuk-form-group--error' css class.
+  #
+  # Passing the attribute symbol will only add the error class if the object contains errors for
+  # the matching symbol.
+  #
+  # Generally pass only an object when wrapping multiple components and pass an object and attribute when
+  # wrapping a single component.
+  #
+  # === Examples
+  # ==== Wrap multiple components
+  #
+  #   <%= form_group_tag object do %>
+  #     <fieldset class="govuk-fieldset">
+  #       <legend class="govuk-fieldset__legend govuk-fieldset__legend--xl">
+  #         <h1 class="govuk-fieldset__heading">
+  #           Where do you live?
+  #         </h1>
+  #       </legend>
+  #     ...
+  #     </fieldset>
+  #   <% end %>
+  #
+  #   <div class="govuk-form-group">
+  #     <fieldset class="govuk-fieldset">
+  #       <legend class="govuk-fieldset__legend govuk-fieldset__legend--xl">
+  #         <h1 class="govuk-fieldset__heading">
+  #           Where do you live?
+  #         </h1>
+  #       </legend>
+  #        ...
+  #     </fieldset>
+  #   </div>
+  #
+  # ==== Wrap a single component
+  #
+  #   <%= form_group_tag object, :event_name do %>
+  #     ...
+  #   <% end %>
+  #
+  #   <div class="govuk-form-group">
+  #     <label class="govuk-label" for="object_event_name">
+  #       Event name
+  #     </label>
+  #     <input class="govuk-input" id="object_event_name" name="event_name" type="text">
+  #   </div>
+  #
+  def form_group_tag(object, attribute = nil, &block)
+    css_classes = ["govuk-form-group"]
+    case attribute
```

Comment:
Given there is only a single branch is this `case` I'd be tempted to write this as an `if` instead. Maybe even without the nested conditionals too:

```ruby
if attribute.present? && object.errors.key?(attribute)
  css_classes += ["govuk-form-group--error"]
elsif object.errors.any?
  css_classes += ["govuk-form-group--error"]
end
```

Reply:
Much better, thanks. Ended up here:

```ruby
if attribute.present? && object.errors.key?(attribute)
  css_classes << "govuk-form-group--error"
elsif attribute.nil? && object.errors.any?
  css_classes << "govuk-form-group--error"
end
```

Reply:
Is `attribute.nil?` necessary? What should happen if e.g. `""` is passed in as the `attribute`?

Reply:
I set `attribute` to `nil` by default (line 56). If you pass "" there would be no match and the error css class would not get added, which is my expected behaviour.

The idea was that not passing `attribute` implies you are using the form group for the whole 'question' (e.g. fullname) whereas passing `attribute` implies use in a multi-field form (e.g. address).

I did have two methods `form_group_tag` and `field_group_tag` perhaps that would be more explicit?

Reply:
I guess I'm just trying to work out what a `blank?` attribute that wasn't `nil` would mean, but honestly, it's unlikely to happen anyway.